src/Package/Impl/ProjectSystem/RProjectLoadHooks.cs

```diff
@@ -207,12 +243,18 @@ internal sealed class RProjectLoadHooks {
             }
         }
 
+#if VS14
         private async Task<bool> GetSaveDefaultWorkspace(string rdataPath) {
+#else
+        private bool GetSaveDefaultWorkspace(string rdataPath) {
+#endif
             switch (_toolsSettings.SaveRDataOnProjectUnload) {
                 case YesNoAsk.Yes:
                     return true;
                 case YesNoAsk.Ask:
+#if VS14
```

Comment:
since most of the code is UI thread (starting with the package load) I added 'requires UI thread' to the AutoLoad attribute. If you think it is worth keeping first 2 lines on bg thread, we can keep code as is